src/matchers/toMatchText/index.ts

```diff
@@ -0,0 +1,29 @@
+import { SyncExpectationResult } from "expect/build/types"
+import { getElementText, getMessage, InputArguments } from "../utils"
+
+const toMatchText: jest.CustomMatcher = async function (
+  ...args: InputArguments
+): Promise<SyncExpectationResult> {
+  try {
+    const { elementHandle, expectedValue } = await getElementText(...args)
+    /* istanbul ignore next */
+    const actualTextContent = await elementHandle.evaluate(
+      (el) => el.textContent
+    )
+    const regex = expectedValue
+    const res = actualTextContent?.match(regex) || []
+
+    return {
+      pass: res.length > 0,
```

Comment:
You could just check that `res` is truthy since it will return `null` if there are no matches or `undefined` if `actualTextContent` returns undefined.  Would also remove the need for the `|| []`

Reply:
This needs to be re-thinked in case of not.match. So I am working on a different approach